python/gherkin/scripts/generate_tokens.py

```diff
@@ -0,0 +1,19 @@
+import sys
+
+from gherkin.token_scanner import TokenScanner
+from gherkin.token_formatter_builder import TokenFormatterBuilder
+from gherkin.parser import Parser
+
+
+def main() -> int:
+    files = sys.argv[1:]
+    parser = Parser(TokenFormatterBuilder())
+    for file in files:
+        scanner = TokenScanner(file)
+        print(parser.parse(scanner))
+
+    return 0
+
+
+if __name__ == "__main__":
+    raise SystemExit(main())
```

Comment:
BAD IDEA: `raise SystemExit(main())` — Raising an exception and calling the `main()` function as side-effect
* Use `sys.exit(main())` instead

Reply:
`sys.exit(...)` and `raise SystemExit(...)` are completely equivalent, so this is only about style.

I prefer the latter just because it avoids an import. Wdyt?

Reply:
> calling the main() function as side-effect

This is not really a side-effect. You may argue that also in `sys.exit(main())` it is a side effect.

Reply:
Agreed this is mainly a styling thing from what I can see, unless I am mistaken?

Reply:
Actually, it is not. It is the difference between using the official API for something and **programming by side-effects**.
* Show me one example in https://docs.python.org where your idiom is used in the context of a `__main__` section
* Show me one example why packaging generates scripts the way I describe instead of how you use it

SEE ALSO:
* https://docs.python.org/3/library/__main__.html

Reply:
How is this programming by side effect? Where is the side effect here?

Reply:
anyway, I removed `raise SystemExit` in 5405b62f to make everyone happy